File: java/behavioral_pattern/strategy/BitCoinPayment.java
package strategy;

public class BitCoinPayment implements PaymentStrategy {
    private String walletAddress;

    public BitCoinPayment(String walletAddress) {
        this.walletAddress = walletAddress;
    }

    @Override
    public void pay(int amount) {
        System.out.println("Paid $" + amount + " using BitCoin");
        System.out.println("Wallet Address: " + walletAddress);
        System.out.println("Transaction confirmed on blockchain");
    }
}

File: typescript/behavioral_pattern/src/strategy/BitCoinPayment.ts
import PaymentStrategy from "./PaymentStrategy";

class BitCoinPayment implements PaymentStrategy {
    private walletAddress: string;

    public constructor(walletAddress: string) {
        this.walletAddress = walletAddress;
    }

    public pay(amount: number): void {
        console.log(`Paid ${amount} using BitCoin.`);
        console.log(`Wallet Address: ${this.walletAddress}`);
        console.log(`Transaction confirmed on blockchain`);
    }
}

export default BitCoinPayment;